simulation/chart.py

```diff
@@ -1,232 +1,438 @@
+            style (PenStyle, optional): Style of the plot line. Defaults to None.
+            autoDownsample (bool, optional): If True, resample the data before plotting to avoid plotting multiple line segments per pixel. Defaults to None.
+            downsample (int,optional): Reduce the number of samples displayed by the given factor. Defaults  to None.
+            downsampleMethod (DownsampleMethod,optional): Method to use for downsampling. Defaults to None.
+        """
+        config = {"name": name}
+        config["pen"] = _make_pen(color=color, width=width, style=style)
+        plot = self.__chart.plot(**config)
+        ChartWidget._update_downsapling(
+            plot, auto=autoDownsample, ds=downsample, method=downsampleMethod
         )
-        .output(file_name, vcodec=codec, pix_fmt=pixel_format)
-        .overwrite_output()
-        .run_async(pipe_stdin=True)
-    )
-    return process
 
+        self.__plots[id] = plot
+
+    def add_scatter_plot(
+        self,
+        id: str,
+        name: str,
+        size: Optional[float] = None,
+        color: Optional[Color] = None,
+        symbol: Optional[SymbolType] = None,
+        border: Optional[PenConfig] = None,
+        autoDownsample: Optional[bool] = None,
+        downsample: Optional[int] = None,
+        downsampleMethod: Optional[DownsampleMethod] = None,
+    ):
+        """Add a scatter plot to the chart.
+
+        Args:
+            id (str): Unique id of the plot in the chart
+            name (str): Name (legend title) of the plot
+            size (float, optional): Size of the marker symbol. Defaults to None.
+            color (Color, optional): Color to fill the marker symbol. Defaults to None.
+            symbol (SymbolType, optional): Shape of the marker symbol. Defaults to None.
+            border (PenConfig, optional): Pen to draw border around the marker symbol. Defaults to None.
+            autoDownsample (bool, optional): If True, resample the data before plotting to avoid plotting multiple line segments per pixel. Defaults to None.
+            downsample (int,optional): Reduce the number of samples displayed by the given factor. Defaults  to None.
+            downsampleMethod (DownsampleMethod,optional): Method to use for downsampling. Defaults to None.
+        """
+        config = {"name": name, "pen": None}
+
+        if size is not None:
+            config["symbolSize"] = size
+
+        if color is not None:
+            config["symbolBrush"] = _make_color(color)
+
+        if symbol is not None:
+            config["symbol"] = symbol
+
+        if border is not None:
+            config["symbolPen"] = _make_pen(**border)
+
+        plot = self.__chart.plot(**config)
+        ChartWidget._update_downsapling(
+            plot, auto=autoDownsample, ds=downsample, method=downsampleMethod
+        )
+
+        self.__plots[id] = plot
+
+    def set_data(self, id: str, *args, **kargs):
```

Comment:
Normally, in python, `args` and `kwargs`. You're missing the `w`.

Reply:
https://github.com/semiotic-ai/autoagora-agents/commit/543ffd999da9d9e71b97c69fd84b59c0dec6a395